lib/ProductOpener/Food.pm

```diff
@@ -1428,6 +1577,107 @@ sub compute_nutrition_score ($product_ref) {
 	return;
 }
 
+=head2 set_fields_comparing_nutriscore_versions($product_ref, $version1, $version2)
+
+When we are migrating from one version of the Nutri-Score to another (e.g. 2021 vs 2023),
+we may compute both version for a time. This function sets temporary fields to ease the comparison
+of both versions.
+
+Once the migration is complete, those fields will no longer be computed.
+
+=cut
+
+sub set_fields_comparing_nutriscore_versions ($product_ref, $version1, $version2) {
+
+	my $nutriscore1 = $product_ref->{nutriscore}{$version1}{grade};
+	my $nutriscore2 = $product_ref->{nutriscore}{$version2}{grade};
+
+	# Set tags fields for both versions
+	$product_ref->{"nutriscore_${version1}_tags"} = [$nutriscore1];
+	$product_ref->{"nutriscore_${version2}_tags"} = [$nutriscore2];
+
+	# Compare both versions
+	if ($nutriscore1 eq $nutriscore2) {
+		push @{$product_ref->{misc_tags}}, "en:nutriscore-$version1-same-as-$version2";
+	}
+	elsif ($nutriscore1 lt $nutriscore2) {
+		push @{$product_ref->{misc_tags}}, "en:nutriscore-$version1-better-than-$version2";
+	}
+	else {
+		push @{$product_ref->{misc_tags}}, "en:nutriscore-$version1-worse-than-$version2";
+	}
+
+	push @{$product_ref->{misc_tags}}, "en:nutriscore-$version1-$nutriscore1-$version2-$nutriscore2";
+
+	return;
+}
+
+=head2 compute_nutriscore( $product_ref )
+
+Determines if we have enough data to compute the Nutri-Score (category + nutrition facts),
+and if the Nutri-Score is applicable to the product the category.
+
+Populates the data structure needed to compute the Nutri-Score and computes it.
+
+=cut
+
+sub compute_nutriscore ($product_ref, $current_version = "2021") {
+
+	# Initialize values
+
+	$product_ref->{nutrition_score_debug} = '';
+
+	# Remove any previously existing Nutri-Score related fields
+	remove_nutriscore_fields($product_ref);
+
+	my $nutriscore_applicable = is_nutriscore_applicable_to_the_product_categories($product_ref);
+
+	my ($nutrients_available, $prepared, $nutriments_field)
+		= check_availability_of_nutrients_needed_for_nutriscore($product_ref);
+
+	if (not($nutriscore_applicable and $nutrients_available)) {
+		push @{$product_ref->{misc_tags}}, "en:nutriscore-not-computed";
```

Comment:
we use the same tag for not applicable and not available ? Is'nt it a pity ?

Reply:
we also add other misc tags for those separately, in the functions that compute them